### PR TITLE
feat: drop `v2g` and reimplement distance features

### PR DESCRIPTION
**Pull request by ireneisdoomed**
## ✨ Context

This PR closes [3434](https://github.com/opentargets/issues/issues/3434) and [3258](https://github.com/opentargets/issues/issues/3258)

The definition of the features have been implemented as agreed with @addramir and @xyg123*. The spreadsheet that documents them has been updated accordingly https://docs.google.com/spreadsheets/d/1wUs1AprRCCGItZmgDhc1fF5BtwCSosdzFv4NQ8V6Dtg/edit?gid=452826388#gid=452826388

## 🛠 What does this PR implement

- Removal of V2G as a step and a dataset.
- We still have the intervals dataset, but we are not dumping it (will be processed by L2G)
- Generation of distance features based on the variant index, not V2G
- Generation of negative gold standard based on the variant index, not V2G
- Addition of distance to footprint features
- Addition of all features in the neighbourhood
- Testing

## 🙈 Missing

- Discuss the filtering on protein coding genes when calculating the neighbourhood features. To be implemented in another PR @addramir 

## 🚦 Before submitting

- [ ] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [x] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

**Comment by d0choa** [comment]
The new features are definitely a step forward to make it more modular and easy to interpret.

File: src/gentropy/dataset/l2g_feature.py [hunk 1]
```diff
@@ -395,95 +768,471 @@ def compute(
...
+            how="inner",
+        )
+        .withColumn(
+            "weightedDistance",
+            f.col(distance_type) * f.col("variantInLocusPosteriorProbability"),
+        )
+        .groupBy("studyLocusId", "geneId")
+        .agg(agg_expr.alias(feature_name))
+    )
+
```
Comment (xyg123) [inline src/gentropy/dataset/l2g_feature.py:874]: Are we using weighted distances as the feature now? instead of the normalised distance score between 0-1?

**Review by xyg123 — COMMENTED** [review]
Looks really nice, good bye v2g, just need to make sure we are happy with the weighted distance features. x

File: src/gentropy/dataset/intervals.py [hunk 2]
```diff
@@ -74,32 +71,3 @@ def from_source(
         source_class = source_to_class[source_name]
         data = source_class.read(spark, source_path)  # type: ignore
         return source_class.parse(data, gene_index, lift)  # type: ignore
```
Comment (xyg123) [inline src/gentropy/dataset/intervals.py:73]: Just wondering, what is the plan with the interval dataclass?  Do we plan to have a separate DAG to process these and save them somewhere?

File: src/gentropy/dataset/l2g_feature.py (hunk 1 above)
Comment (ireneisdoomed) [reply src/gentropy/dataset/l2g_feature.py:874]: We used to weight them earlier as well. The feature itself is not normalised until later.

**Review by xyg123 — APPROVED** [review]
Everything looks good!

**Pull request merged**